server/src/main/java/org/eclipse/lsp/cobol/core/MessageServiceParser.java

```diff
@@ -182,8 +207,14 @@ private Integer tryParseInt(String input) {
 
   private Boolean checkStartsWith(String input, String[] startsWith) {
     for (String item : startsWith) {
-      if (input.startsWith(item))
-        return true;
+      if (input.startsWith(item)) return true;
+    }
+    return false;
+  }
+
+  private Boolean checkInputInAllowedValues(String input, String[] allowedValues) {
+    for (String item : allowedValues) {
+      if (input.equals(item)) return true;
```

Comment:
I guess that allowed values are not case-sensitive. Can you convert the `input` to the upper case?

Reply:
`equalsIgnoreCase` should also work. Also, the best practice is to compare the thing we are sure is not null (allowedValues here) with ones we are not sure (user input). It helps to avoid the NullPointerException.
```suggestion
      if (item.equalsIgnoreCase(input)) return true;
```